platform/legacy/LegacyPlatformConfig.ts

```diff
@@ -27,31 +31,32 @@ interface LegacyLoggingConfig {
  * supported by the current platform.
  */
 export class LegacyConfigToRawConfigAdapter implements RawConfig {
-  constructor(private readonly legacyConfig: LegacyConfig) {}
+  private newPlatformConfig: ObjectToRawConfigAdapter;
+
+  constructor(private readonly legacyConfig: LegacyConfig) {
+    this.newPlatformConfig = new ObjectToRawConfigAdapter({
+      [NEW_PLATFORM_CONFIG_ROOT]:
```

Comment:
__note:__ having `{__newPlatform: .... }` here instead of `{....}` is just easier since we won't have to strip `__newPlatform` prefix for every config call.